tracing_test.go

```diff
@@ -60,23 +65,47 @@ func TestTracingSpanAttributes(t *testing.T) {
 		DisableRelay:   true,
 	}
 	WithVerifiedServer(t, opts, func(ch *Channel, hostPort string) {
+		const (
+			customAppHeaderKey           = "futurama"
+			customAppHeaderExpectedValue = "simpsons"
+		)
+		var customAppHeaderValue atomic.Value
 		// Register JSON handler
-		jsonHandler := &JSONHandler{TraceHandler: testtracing.TraceHandler{Ch: ch}, t: t}
+		jsonHandler := &JSONHandler{
+			TraceHandler: testtracing.TraceHandler{Ch: ch},
+			t:            t,
+			sideEffect: func(ctx json.Context) {
+				customAppHeaderValue.Store(ctx.Headers()[customAppHeaderKey])
+			},
+		}
 		json.Register(ch, json.Handlers{"call": jsonHandler.callJSON}, jsonHandler.onError)
 
 		span := ch.Tracer().StartSpan("client")
 		span.SetBaggageItem(testtracing.BaggageKey, testtracing.BaggageValue)
 		ctx := opentracing.ContextWithSpan(context.Background(), span)
 		root := new(testtracing.TracingResponse).ObserveSpan(ctx)
 
+		// Pretend that the client propagated tracing headers from upstream call, and test that the outbound call
+		// will override them (https://github.com/uber/tchannel-go/issues/682).
+		headers := make(map[string]string)
+		assert.NoError(t, tracer.Inject(span.Context(), opentracing.TextMap, opentracing.TextMapCarrier(headers)))
+		tracingHeaders := map[string]string{
+			customAppHeaderKey: customAppHeaderExpectedValue,
+		}
+		for k, v := range headers {
+			tracingHeaders["$tracing$"+k] = v
```

Comment:
can you set the value to something incorrect, like `dummy` so we know that it was replaced